code/datums/gamemodes/spy_theft.dm

```diff
@@ -341,151 +341,172 @@
+		big_station_bounties[/obj/machinery/computer/card] = 2
+		big_station_bounties[/obj/machinery/computer/aiupload] = 3
+		big_station_bounties[/obj/machinery/computer/genetics] = 2
+		big_station_bounties[/obj/machinery/computer/supplycomp] = 1
+		big_station_bounties[/obj/machinery/computer/robotics] = 2
+		big_station_bounties[/obj/machinery/power/reactor_stats] = 1
+
+		big_station_bounties[/obj/machinery/computer3/generic/communications] = 1
+		big_station_bounties[/obj/machinery/computer3/terminal/zeta] = 1
+		big_station_bounties[/obj/machinery/vending/security] = 2
+		big_station_bounties[/obj/machinery/vending/medical] = 1
+		big_station_bounties[/obj/machinery/vending/port_a_nanomed] = 1
+		big_station_bounties[/obj/machinery/vending/fortune] = 1
+		big_station_bounties[/obj/machinery/vending/standard] = 1
+
+		big_station_bounties[/obj/machinery/port_a_brig] = 3
+		big_station_bounties[/obj/machinery/flasher/portable] = 2
+		big_station_bounties[/obj/machinery/sleeper/port_a_medbay] = 1
+		big_station_bounties[/obj/machinery/atmospherics/unary/cryo_cell] = 2
+		big_station_bounties[/obj/machinery/computer/cloning] = 2
+		big_station_bounties[/obj/machinery/clonepod] = 2
+		big_station_bounties[/obj/machinery/clonegrinder] = 1
+		big_station_bounties[/obj/machinery/genetics_scanner] = 2
+
+		big_station_bounties[/obj/machinery/recharge_station] = 1
+		big_station_bounties[/obj/machinery/optable] = 2
+		big_station_bounties[/obj/storage/closet/port_a_sci] = 2
+
+		big_station_bounties[/obj/machinery/manufacturer/robotics] = 1
+		big_station_bounties[/obj/machinery/manufacturer/medical] = 1
+		big_station_bounties[/obj/machinery/manufacturer/general] = 1
+		big_station_bounties[/obj/machinery/manufacturer/general] = 1
+
+		big_station_bounties[/obj/submachine/chef_oven] = 1
+		big_station_bounties[/obj/machinery/gibber] = 1
+
+		big_station_bounties[/obj/machinery/bot/guardbot] = 1
+		big_station_bounties[/obj/machinery/plantpot] = 1
+		big_station_bounties[/obj/machinery/partyalarm] = 1
+		big_station_bounties[/obj/pool_springboard] = 1
+
+		big_station_bounties[/obj/reagent_dispensers] = 2
+		big_station_bounties[/obj/machinery/crusher] = 2
+		big_station_bounties[/obj/machinery/communications_dish] = 2
 
 	active_bounties.len = 0
 
+	// Find matches for all our possible bounties
+	var/list/spy_thief_target_types = list()
+	// Construct master list of target types
+	for(var/i=1, i<=station_bounties.len, i++)
+		spy_thief_target_types += station_bounties[i]
+	for(var/i=1, i<=big_station_bounties.len, i++)
+		spy_thief_target_types += big_station_bounties[i]
+	var/list/valid_spy_thief_targets_by_type = list()
+	// Loop through all objects and pick valid objects on station
+	for(var/obj/Object in world)
+		LAGCHECK(LAG_LOW)
+		var/turf/Turf = get_turf(Object)
+		if(!Turf || Turf.z != Z_LEVEL_STATION)
+			continue
+		var/area/A = get_area(Object)
+		if (A.name == "Listening Post")
+			continue
+		for(var/type in spy_thief_target_types)
```

Comment:
Should just be able to `check spy_thief_target_types[Object.type]` instead of iterating.  Should also be one of the first checks of the for loop because it is FAST.

Reply:
This sorta depends on if you want to include subtypes or not. The behaviour is certainly not identical.

Reply:
I'd probably split it maybe into two sets of types based on which thing you want to do.

Reply:
Added this as a double check.
It's a bit annoying, I think the only places we use subtypes are:
/obj/item/instrument
/obj/item/storage/backpack
/obj/machinery/vehicle
/obj/machinery/computer/card
/obj/item/storage/backpack
/obj/item/device/radio/headset/command

Most of these could turn into 4~ individual entries but /obj/machinery/computer/card and /obj/item/instrument would be a lot longer. It might actually be worth putting in the subtypes if the speed is that worthwhile. It makes not picking a repetitive list harder but that could be solved by picking one or two random entries from the subtype rather than using all of them.